compiler/backend/src/org/jetbrains/kotlin/codegen/inline/LambdaInfo.kt

```diff
@@ -130,84 +100,52 @@ abstract class DefaultLambda(
     final override lateinit var invokeMethod: Method
         private set
 
-    private lateinit var invokeMethodDescriptor: FunctionDescriptor
-
-    override val invokeMethodParameters: List<KotlinType?>
-        get() = invokeMethodDescriptor.valueParameters.map { it.returnType } // should be FunctionN, so no extension receiver
-
-    override val invokeMethodReturnType: KotlinType?
-        get() = invokeMethodDescriptor.returnType
-
     final override lateinit var capturedVars: List<CapturedParamDesc>
         private set
 
     var originalBoundReceiverType: Type? = null
         private set
 
-    override val isSuspend = parameterDescriptor.isSuspendLambda
+    override val isSuspend = false // TODO: it should probably be true sometimes, but it never was
```

Comment:
FYI: Frontend reports: "Suspend functional parameters with default values are not yet supported in inline functions"

Reply:
Could you mention it here?

Reply:
Huh, it really does have this error, but the `functionDescriptor.isSuspend` check means that it's only reported on `suspend inline fun f(x: suspend () -> Unit = {})` and not `inline fun f(crossinline x: suspend () -> Unit = {})` even though both are broken. Using suspend *function references* works though, the problem is specifically with suspend lambdas (there's no version of `invoke` without the state machine, so they're completely uninlinable).

Reply:
OK, thank you for detailed explanation